File: check_db.js
const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');

const config = fs.readFileSync('supabase.js', 'utf8');
const urlMatch = config.match(/supabaseUrl\s*=\s*['"]([^'"]+)['"]/);
const keyMatch = config.match(/supabaseKey\s*=\s*['"]([^'"]+)['"]/);

if (!urlMatch || !keyMatch) {
    console.log("Could not find supabase credentials");
    process.exit(1);
}

const supabase = createClient(urlMatch[1], keyMatch[1]);

async function checkData() {
    const { data: tx, error: err1 } = await supabase.from('transactions').select('*').eq('reference_number', '9678735915').single();
    console.log("Transaction:", tx);
    
    const { data: mapping, error: err2 } = await supabase.from('item_mappings').select('*').ilike('item_name', '%Spring 26 Fees%');
    console.log("Mappings:", mapping);
}
checkData();

File: check_db.py
import requests
import json

URL = 'https://hjtxdyuevxcezxzbiiqk.supabase.co/rest/v1'
KEY = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImhqdHhkeXVldnhjZXp4emJpaXFrIiwicm9sZSI6ImFub24iLCJpYXQiOjE3Nzk1MjU5MDEsImV4cCI6MjA5NTEwMTkwMX0.ZiKUw1db5pDRYto-hLGut3rdrzxVfRN36ouX4AjB5AQ'
HEADERS = {
    'apikey': KEY,
    'Authorization': f'Bearer {KEY}'
}

# Fetch transaction
res_tx = requests.get(f'{URL}/transactions?reference_number=eq.9678735915', headers=HEADERS)
tx_data = res_tx.json()
print("Transaction Data:")
print(json.dumps(tx_data, indent=2))

# Fetch mapping for Spring 26 Fees
res_map = requests.get(f'{URL}/item_mappings?item_name=ilike.*Spring%2026%20Fees*', headers=HEADERS)
map_data = res_map.json()
print("\nMapping Data:")
print(json.dumps(map_data, indent=2))

